### PR TITLE
chore: typecheck examples during CI

### PR DESCRIPTION
**Pull request by fiam**
Don't just execute them, always validate the types

### CODE REVIEW

File: scripts/test-examples/single.sh [hunk 1]
```diff
@@ -103,6 +99,17 @@ if grep -q '"test:playwright"' package.json; then
     WG_NODE_URL=${default_node_url} npm run test:playwright
 fi
 
+# If the example uses Next.js, compile it
+if grep -q '"build:next"' package.json; then
+    # This example doesn't build under a pnpm workspace
+    if !grep -q '"nextjs-react-query"' package.json; then
+        npm run build:next
+    fi
+elif grep -q '"check"' package.json; then
```
Comment (StarpTech) [inline scripts/test-examples/single.sh:108]: `check` should be part of the build script. We should split it.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: scripts/test-examples/single.sh (hunk 1 above)
Comment (fiam) [reply scripts/test-examples/single.sh:108]: All examples have `build` call `wunderctl generate`, while `check` calls `tsc --noEmit`. We've had that for both examples and testapps for quite long. Are you suggesting we change all those `package.json` to make `build` also call `check?

Comment (fiam) [reply scripts/test-examples/single.sh:108]: I was not aware of those plans. So should use the following commands in `package.json`: `"generate": "wunderctl generate"`, `"build": "npm run generate && npm run check"`?

Comment (fiam) [reply scripts/test-examples/single.sh:108]: I was not aware of those plans. So should we add `"build:wundergraph": "wunderctl generate"` and `"build": "build:wundergraph && npm run check`

Comment (fiam) [reply scripts/test-examples/single.sh:108]: No idea how the double comment happened, apologies for the inconvenience!

Comment (StarpTech) [reply scripts/test-examples/single.sh:108]: Not set in stone. I'd make tsc check part of `build`.

Comment (fiam) [reply scripts/test-examples/single.sh:108]: As discussed, we're splitting that into a separate issue. Let's concentrate on fixing type-checking in the examples for this one. PTAL.

**Review by StarpTech — APPROVED** [review]
LGTM

**Review by Aenimus — APPROVED** [review]

**Pull request merged**